frontend/src/metabase-types/forms/index.ts

```diff
@@ -68,6 +68,10 @@ export type FormField<Value = DefaultFieldValue> = {
   onUpdate: () => void;
 };
 
+export type NestedFormField = {
+  [key: string]: FormField | NestedFormField;
+};
```

Comment:
The actual field type of `fields` which could be nested e.g. database form.
![image](https://user-images.githubusercontent.com/1937582/170470023-b6930b35-6201-4196-9b8f-39c054ee6d7d.png)